modules/hostap/CMakeLists.txt

```diff
@@ -398,6 +398,7 @@ zephyr_library_compile_definitions_ifdef(CONFIG_EAP_TLS
 
 zephyr_library_sources_ifdef(CONFIG_EAP_TTLS
   ${HOSTAP_SRC_BASE}/eap_peer/eap_ttls.c
+  ${HOSTAP_SRC_BASE}/eap_common/chap.c
```

Comment:
to avoid adding twice, maybe we can do something like:

```
if(CONFIG_EAP_TTLS OR CONFIG_EAP_MSCHAPV2)
  zephyr_library_sources( ${HOSTAP_SRC_BASE}/eap_common/chap.c)
endif()
```

Reply:
Hi @krish2718 
TTLS and MSCHAPV2 are separated  EPA security. Should we keep the original config, but not add this change? 
If want to support EAP-TTLS-MSCHAPv2, we should enable both EAP-TTLS macro and EAP-MSCHAPV2 macro, is this will be more reasonable?

Reply:
Sure, but my suggested fix is `a or b`, so, `chap.c` would be includes in that case, no?

Reply:
I think EAP-TTLS is independent with MSCHAPV2.
EAP-TTLS can also combine with other pharse2 EAP type, in this case,  eap_common/chap.c will be build in application, but not used.